client/python/armada_client/client.py

```diff
@@ -21,6 +21,7 @@
 from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
 from armada_client.permissions import Permissions
 from armada_client.typings import JobState
+import tenacity
```

Comment:
I think you will need to add tenacity to our pyproject.toml file as this is a new dependency.  

You will also need to run the python code formatter (Black via `tox -e format-code`) to get your code to pass CI.

Reply:
Done!

Reply:
Can you remove tenacity from the client?  Not sure I like this change being enforced on all users of the client.

I’d leave the changed in jobservice.  Can you also experiment with deploying jobservice with multiple replicas

Reply:
I didnt get you but in the comments of the issue , It was mentioned that we need to include backoff for all the clients(ArmadaClient & JobServiceClient).

Reply:
Thanks for removing the client side.  The main user of this is mostly for jobservice and I'm not sure we want to add retries to all users of the client.

Reply:
Hey @kannon92  , I added the test for tenacity.